python/dask_cudf/dask_cudf/io/csv.py

```diff
@@ -24,7 +24,15 @@ def _internal_read_csv(path, chunksize="256 MiB", **kwargs):
     if isinstance(chunksize, str):
         chunksize = parse_bytes(chunksize)
 
-    filenames = sorted(glob(str(path)))
+    if isinstance(path, list):
+        filenames = path
+    elif isinstance(path, str):
+        filenames = sorted(glob(path))
+    elif hasattr(path, "__fspath__"):
+        filenames = sorted(glob(path.__fspath__()))
+    else:
+        raise TypeError("Path type not understood:{}".format(type(path)))
```

Comment:
Should we turn this into a function and call that function from each reader? Does Dask have a utility that does this already so we're not reinventing the wheel?

Reply:
`fsspec` might have something?  @martindurant

Reply:
`fsspec.open_files` ? I'm not sure what the intent is here: are you wanting to only support local paths and Path objects?
Allowing anything with `fspath` may not be helpful, since this includes already open files too.

Reply:
> Should we turn this into a function and call that function from each reader?

We could make this a utility but right now `csv` and `orc` are the only two readers that need this. `parquet` and `json` use a different approach.

> `fsspec.open_files` ? I'm not sure what the intent is here: are you wanting to only support local paths and Path objects?
> Allowing anything with `fspath` may not be helpful, since this includes already open files too.

This is an alternate codepath for local paths and path objects. `fsspec.open_files` could work but would return file objects. The code would just use the filenames from those objects and pass those down. `fsspec.expand_paths_if_needed` might be another alternative but would require creating a local fs object and passing that as an argument to the method.

I am in favor of keeping the current implementation for now and revisiting this in the future if `dask_cudf.io` is refactored to be more generalized. Open to making changes if one of you feels otherwise.

Reply:
Note that `open_files` returns our own `OpenFile` instances, and does not directly open the target file(s) until used within a `with` block. That may or may not be helpful to you.